### PR TITLE
Add limits module

### PR DESCRIPTION
**Pull request by konan-abhi**
E.g. used in keystone service to register limits for particular service.

This includes gophercloud version bump to 1.4.0

### CODE REVIEW

**Comment by fmount** [comment]
+1 from me, let's merge as long as the CI is green (I suspect we should fix `pre-commit` tests)

**Comment by konan-abhi** [comment]
> +1 from me, let's merge as long as the CI is green (I suspect we should fix `pre-commit` tests)

Any inputs on how to fix those?

**Comment by fmount** [comment]
> > +1 from me, let's merge as long as the CI is green (I suspect we should fix `pre-commit` tests)
> 
> Any inputs on how to fix those?

on your local branch, run `pre-commit run --all-files` and `git add` the resulting changes.
With that it should work on the next iteration!

**Comment by konan-abhi** [comment]
> pre-commit run --all-files

Thank you!

**Review by fmount — APPROVED** [review]
+1

File: modules/openstack/go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.19
 
 require (
 	github.com/go-logr/logr v1.2.4
-	github.com/gophercloud/gophercloud v1.2.0
+	github.com/gophercloud/gophercloud v1.4.0
```
Comment (fmount) [inline modules/openstack/go.mod:7]: @stuggi FYI bumping gophercloud to 1.4.0 to get [1] in.

[1] https://github.com/gophercloud/gophercloud/pull/2616

Comment (abays) [reply modules/openstack/go.mod:7]: I would say it's fine if everything still works :)

**Pull request merged**